src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.RoleDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.model.Role;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+/**
+ * @author Sergey Klunniy
+ */
+@Repository
+public class RoleDaoImpl implements RoleDao {
+
+    private SessionFactory sessionFactory;
+
+    public RoleDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Role addRole(Role role) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(role);
+            transaction.commit();
+            role.setId(id);
+            return role;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("can't add role to db", e);
```

Comment:
```suggestion
            throw new DataProcessingException("can't add role to db", e);
```
@MateAcademy давай будемо притримуватися одного стилю написання коду

Reply:
змінив на DataProcessingException, я просто пропустив цей момент